core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -332,7 +332,7 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
     // Member id must have formatted with valid delimiter as [prefix]-[suffix] to
     // be validated.
     if (delimiterIdx == -1) {
-      info(s"given member.id $memberId was ill-formatted without delimiter '-'.")
+      error(s"given member.id $memberId was ill-formatted without delimiter '-'.")
```

Comment:
I'm actually thinking we should just reject the request all together if its member-id is ill-formatted (of course, not in this function then, but rather do that at the very beginning of every request handling). But if you feel it is good enough with sound rationales, I'm fine to leave it as is.

Reply:
@guozhangwang yea the reasoning is like we need to do a lot of unit test refactoring in this case. I could do that in a separate PR.

Reply:
Sounds good. I agree it may be out of the scope for this PR.